tests/templates/test_embeddings/test_amplitude.py

```diff
@@ -0,0 +1,352 @@
+                features=x, wires=range(n_qubits), pad_with=0.0, normalize=False
+            )
+            return qml.expval(qml.PauliZ(0))
+
+        with pytest.raises(ValueError, match="Features must be of length"):
+            circuit(x=inpt)
+
+    def test_amplitude_embedding_tolerance_value(self):
+        """Tests that a small enough tolerance value is used for Amplitude
+        Embedding."""
+        inputs = np.array(
+            [
+                0.25895178024895,
+                0.115997030111517,
+                0.175840500169049,
+                0.16545033015906,
+                0.016337370015706,
+                0.006616800006361,
+                0.22326375021464,
+                0.161815530155566,
+                0.234776190225708,
+                0.082623190079432,
+                0.291982110280705,
+                0.295344560283937,
+                0.05998731005767,
+                0.056911140054713,
+                0.274260680263668,
+                0.163596590157278,
+                0.048460970046589,
+                0.292306260281016,
+                0.292451040281155,
+                0.007849840007547,
+                0.218302930209871,
+                0.326763300314142,
+                0.163634550157314,
+                0.275472160264832,
+                0.105510810101436,
+            ]
+        )
+
+        tolerance = 10e-10
+        num_qubits = 5
+        dev = qml.device("default.qubit", wires=num_qubits)
+        assert np.isclose(np.sum(np.abs(inputs) ** 2), 1, tolerance)
+
+        @qml.qnode(dev)
+        def circuit(x=None):
+            qml.templates.AmplitudeEmbedding(
+                x, list(range(num_qubits)), pad_with=0.0, normalize=True
+            )
+            return qml.expval(qml.PauliZ(0))
+
+        # No normalization error is raised
+        circuit(x=inputs)
+
+    def test_deprecated_pad_arg(self):
+        """Test that the pad argument raises a deprecation warning"""
+
+        num_qubits = 2
+        dev = qml.device("default.qubit", wires=num_qubits)
+        inputs = np.array([1.0, 0.0, 0.0, 0.0])
+
+        @qml.qnode(dev)
+        def circuit(x=None):
+            qml.templates.AmplitudeEmbedding(x, list(range(num_qubits)), pad=0.0, normalize=True)
+            return qml.expval(qml.PauliZ(0))
+
+        with pytest.warns(
+            PendingDeprecationWarning,
+            match="will be replaced by the pad_with option in future versions",
+        ):
+            circuit(x=inputs)
+
+
+def circuit_template(features):
+    qml.templates.AmplitudeEmbedding(features, wires=range(3))
+    return qml.state()
+
+
+def circuit_decomposed(features):
+    # need to cast to complex tensor, which is implicitly done in the template
+    qml.QubitStateVector(qml.math.cast(features, np.complex128), wires=range(3))
+    return qml.state()
+
+
+class TestInterfaces:
+    """Tests that the template is compatible with all interfaces."""
```

Comment:
Note: we do not guarantee differentiability for this template, which is why gradients are not tested...